ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -437,6 +442,18 @@ public void replicate() {
                   new ExchangeMetadataResponse(missingStoreKeys, replicaMetadataResponseInfo.getFindToken(),
                       replicaMetadataResponseInfo.getRemoteReplicaLagInBytes());
               exchangeMetadataResponseList.add(exchangeMetadataResponse);
+              // update replication lag in ReplicaSyncUpService
+              if (replicaSyncUpService != null) {
+                ReplicaId localReplica = remoteReplicaInfo.getLocalReplicaId();
+                ReplicaId remoteReplica = remoteReplicaInfo.getReplicaId();
+                boolean updated = replicaSyncUpService.updateLagBetweenReplicas(localReplica, remoteReplica,
+                    exchangeMetadataResponse.localLagFromRemoteInBytes);
+                if (updated && replicaSyncUpService.isSyncUpComplete(localReplica)) {
+                  // complete BOOTSTRAP -> STANDBY transition
+                  remoteReplicaInfo.getLocalStore().setCurrentState(ReplicaState.STANDBY);
```

Comment:
I'm confused - how do we know this replica was in bootstrap state before the check?  Couldn't it be primary?

Reply:
As discussed offline, whether it's a bootstrap replica is determined by `updated` boolean value. I will add comment to clarify this.

Reply:
Minor: If isSyncUpComplete() can throw an expected IllegalStateException, the error code below should not be UnknownError.  Unless we lost track of why the bootstrap failed, in which case it might be okay.

Reply:
I tried to separate IllegalStateException from Exception for this case but then I found `catch (Exception e)` capture any exception from `getMissingStoreKeys`, `processReplicaMetadataResponse` and `isSyncUpComplete` and set error code to a general `Unknown_Error`. I guess previous author of this code combined all exception cases into single error code and drove us dig into the root cause. The cause is actually logged in this catch block:
```
logger.error(
                  "Remote node: " + remoteNode + " Thread name: " + threadName + " Remote replica: " + remoteReplicaInfo
                      .getReplicaId(), e);```
So, I temporarily leave it as it is, to refactor the exception handling may be worth a separate PR.

Reply:
I think it works if you catch StateTransitionException before Exception, but we can punt this to a future PR.
Even if you just catch Exception, you can have a utility that converts exception to error code.